qiskit/ignis/verification/tomography/fitters/gateset_fitter.py

```diff
@@ -0,0 +1,662 @@
+        return bounds_eq
+
+    def bounds_ineq_constraint(self, x: np.array) -> List[float]:
+        """Inequality MLE constraints on the GST data
+
+            Args:
+                x: The vector representation of the GST data
+
+            Returns:
+                The list of computed constraint values (should be >= 0)
+
+            Additional information:
+                We have the following constraints on the GST data, due to
+                the PTM representation we are using:
+                1) Every row of G except the first has entries in [-1,1]
+
+                We implement this as two inequalities per entry.
+
+                For additional info, see section 3.5.2 in arXiv:1509.02921
+        """
+        ptm_matrix = self.ptm_matrix_values(x)
+        bounds_ineq = []
+        n = len(self.Gs)
+        d = (2 ** self.qubits)  # rho is dxd and starts at variable d^2
+        ds = d ** 2
+
+        i = 0
+        for _ in range(n):  # iterate over all Gs
+            i += 1
+            for _ in range(ds - 1):
+                i += 1
+            for _ in range((ds - 1) * ds):  # rest of G^k
+                bounds_ineq.append(ptm_matrix[i] + 1)  # G_k[i] >= -1
+                bounds_ineq.append(-ptm_matrix[i] + 1)  # G_k[i] <= 1
+                i += 1
+            for _ in range(ds ** 2):  # the complex part of G^k
+                i += 1
+        return bounds_ineq
+
+    def rho_trace_constraint(self, x: np.array) -> List[float]:
+        """ The constraint Tr(rho) = 1
+            Args:
+                x: The vector representation of the GST data
+
+            Return:
+                The list of computed constraint values (should be equal 0)
+
+            Additional information:
+                We demand real(Tr(rho)) == 1 and imag(Tr(rho)) == 0
+        """
+        trace = self.rho_trace(x)
+        return [trace[0] - 1, trace[1]]
+
+    def constraints(self) -> List[Dict]:
+        """Generates the constraints for the MLE optimization
+
+        Returns:
+            A list of constraints.
+
+        Additional information:
+            Each constraint is a dictionary containing
+            type ('eq' for equality == 0, 'ineq' for inequality >= 0)
+            and a function generating from the input x the values
+            that are being constrained.
+        """
+        cons = []
+        cons.append({'type': 'eq', 'fun': self.rho_trace_constraint})
+        cons.append({'type': 'eq', 'fun': self.bounds_eq_constraint})
+        cons.append({'type': 'ineq', 'fun': self.bounds_ineq_constraint})
+        return cons
+
+    def process_result(self, x: np.array) -> Dict:
+        """Transforms the optimization result to a friendly format
+            Args:
+                x: the optimization result vector
+
+            Returns:
+                The final GST data, as dictionary.
+        """
+        E, rho, G_matrices = self.split_input_vector(x)
+        result = {}
+        result['E'] = E
```

Comment:
E should be returned as an `Operator` object in the standard basis

Reply:
Done in 9a10c52